src/uri.cr

```diff
@@ -175,6 +181,23 @@ class URI
 
   def initialize(@scheme = nil, @host = nil, @port = nil, @path = "", query : String | Params | Nil = nil, @user = nil, @password = nil, @fragment = nil)
     @query = query.try(&.to_s)
+    validate_path
+  end
+
+  private def validate_path
+    if has_authority?
+      unless path.empty? || path.starts_with?("/")
+        raise ArgumentError.new("Invalid URI component: when there is an authority component, path must not start with a slash (`/`)")
```

Comment:
Although I think the logic is what's intended, this message doesn't match: it says "must not" but actually the path _must_ start with a slash (if non-empty). It could be worded as:

> Invalid URI component: when there is an authority component, path must start with a slash (`/`) or be empty

Thanks for doing the legwork to address this so comprehensively 🙇‍♀️

Reply:
```suggestion
        raise ArgumentError.new("Invalid URI component: when there is an authority component, path must start with a slash (`/`) or be empty")
```